common/transcript.go

```diff
@@ -20,21 +22,22 @@ func NewTranscript(label string) *Transcript {
 
 	transcript := &Transcript{
 		state: digest,
+		buff:  bytes.NewBuffer(make([]byte, 0, 1024)),
 	}
 
 	return transcript
 }
 
-func (t *Transcript) AppendMessage(message []byte, label string) {
-	t.state.Write([]byte(label))
-	t.state.Write(message)
+func (t *Transcript) AppendMessage(message []byte, label []byte) {
+	t.buff.Write(label)
+	t.buff.Write(message)
```

Comment:
Instead of doing small writes to the sha256 buffer, we collect the bytes in the buffer.